hyperspy/io_plugins/attolight.py

```diff
@@ -0,0 +1,351 @@
+# -*- coding: utf-8 -*-
+# Copyright 2007-2020 The HyperSpy developers
+#
+# This file is part of  HyperSpy.
+#
+#  HyperSpy is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+#  HyperSpy is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with  HyperSpy.  If not, see <http://www.gnu.org/licenses/>.
+
+#  for more information on the RPL/RAW format, see
+#  http://www.nist.gov/lispix/
+#  and
+#  http://www.nist.gov/lispix/doc/image-file-formats/raw-file-format.htm
+
+import glob
+import os
+import logging
+import numpy as np
+from hyperspy.misc.utils import DictionaryTreeBrowser
+
+_logger = logging.getLogger(__name__)
+
+# Plugin characteristics
+# ----------------------
+format_name = 'HYPCard_AttolightSEM'
+description = 'Reads Cathodoluminescence data from the Attolight SEM system'
+full_support = False
+# Recognised file extension
+file_extensions = ['bin', 'BIN']
+default_extension = 0
+# Writing capabilities
+writes = False
+# ----------------------
+
+# Attolight system specific parameters
+# ------------------------------------
+# Store the system specific parameters in a dictionary for
+# different Attolight SEMs around the world, with their identifier.
+
+attolight_systems = {
+    'cambridge_uk_attolight': {
```

Comment:
Every system has its own calibration parameters, they need to be stored somewhere as they are needed for the loading of the file.
Other readers also have hard coded parameters... 
I don't know where else these could be stored...

Reply:
Good question, there should at least be a way to also specify a local config file. If it is hard coded, new users cannot use the plugin before a PR is through and a new version out. Not sure what the best way would be to do so. However, if we want to provide a way to correct for grating/detector/system response in LumiSpy, we would also have to think of where to store such calibration files without shipping all of them with the package (except maybe some more general ones for common grating blazes or detector models).

Reply:
This still doesn't right to me and needs clarification: either this format is poorly engineered or the PR in this approach is not correct - not expected to be used that way.
By not having other alternative than hard coding the calibration parameter in the reader, I understand that the only attolight software installed in Cambridge can read the calibration data generated in the system in Cambridge (because the calibration is hard coded in the software) and equivalent software installed at other facilities in the world will read the calibration wrongly assuming they will not be aware of the calibration of the Cambrigde system (and because they will use a different calibration parameter specific to their system)?!

Maybe someone from Attolight could advice on this?

> Other readers also have hard coded parameters...

This is not common practice at all to have the calibration parameters hard coded, what reader(s) are you referring to? I can't think of a reader where this is the case.

> I don't know where else these could be stored...

Usually, this information is stored in the file itself or in a separate file, as in the ripple or the empad format. If this is not possible, then, as a default, I would prefer not to calibrate at all with a warning maybe, because not having calibration is better than having a wrong calibration. It will strongly encourage the user to calibrate the data correctly, otherwise a wrong calibration would be misleading. If you can know on what system it has been acquired then you can use your lookup table to set the calibration and it there is not entry for a specific acquisition system, then it should fall back to default (no calibration).

Reply:
A solution could also be to have a default location (specific location in the file system, or same folder as data) where the calibration file should reside. Additionally, the location of the calibration file could be optionally passed as parameter. If no calibration file is found, a message/warning could be printed and it falls back to no calibration.

Then it would be good to add a 'calibrated=True/False' tag to the signal object. Is that compatible with the hyperspy conventions to add such a custom tag @ericpre, @franciscodlp?

Lumispy could then include a routine to separately do the calibration on an uncalibrated signal object (again by providing a calibration file through a default location or with the location passed to the function as parameter).

Reply:
> A solution could also be to have a default location (specific location in the file system [...]) where the calibration file should reside

I see how this could be convenient. However, it is less transparent to the user than the other 2 options, so I think that you should consider carefully if this is worth it.

> [or same folder as data]

This seems to me the best option (after Attolight fixing the format to include the calibration).

> Additionally, the location of the calibration file could be optionally passed as parameter. If no calibration file is found, a message/warning could be printed and it falls back to no calibration.

Sounds excellent. Then, why not simply adding a ``calibration_file`` with a default to a file with a specific name to implement the previous option?

> Then it would be good to add a 'calibrated=True/False' tag to the signal object. Is that compatible with the hyperspy conventions to add such a custom tag @ericpre, @franciscodlp?

What about implementing this by setting ``calibration_file=None``

Do you (@ericpre, @jordiferrero) think that this could be included in v1.6 that we are planning to release today?

Reply:
I have now removed the hard coded values.
Instead, the user can now input a calibration file with the load function providing the 'attolight_calibration_file' parameter.
If no calibration file is provided, then the load function will not calibrate the navigation axis.
I am thinking on maybe removing the calibration for the navigation axis, as it is dependent on many parameters in the metadata... However, it is something that need to be done every time, so for now I am keeping it in the loading function.

I have also removed the calibration of the signal axis as it was not accurate.